contracts/PvpArena.sol

```diff
@@ -467,6 +500,110 @@ contract PvpArena is Initializable, AccessControlUpgradeable {
         return shieldIDs;
     }
 
+    ///@dev function to update the rank of the winner
+    function processWinner(uint256 winnerID) internal {
+        uint256 winnerPoints = characterRankingPoints[winnerID];
+        uint8 tier = getArenaTier(winnerID);
+        uint256[] storage winnerTier = _rankingByTier[tier];
+        uint256 winnerPosition;
+        bool winnerFound;
+
+        // check if winner is withing the top 4
+        for (uint8 i = 0; i < winnerTier.length; i++) {
+            if (winnerID == winnerTier[i]) {
+                winnerPosition = i;
+                winnerFound = true;
+                break;
+            }
+        }
+        // if he is found, compare him to the lower index positions
+        if (winnerFound) {
+            winnerPosition = winnerPosition;
+        }
+        // else, compare it to the 4th one, if he is higher then replace the position and start the loop.
+        else if (
+            winnerPoints >=
+            characterRankingPoints[winnerTier[winnerTier.length - 1]]
+        ) {
+            winnerPosition = winnerTier[winnerTier.length - 1];
+            winnerTier[winnerTier.length - 1] = winnerID;
+        }
+        for (winnerPosition; winnerPosition >= 0; winnerPosition--) {
+            if (winnerPosition <= 0) {
+                break;
+            }
```

Comment:
correct me with this please, I'm breaking the loop is the position is 0 to avoid running the code that comes after. Does that make this more efficient process-wise? or it doesn't make any noticeable change

```
        for (winnerPosition; winnerPosition > 0; winnerPosition--) {
            // break the loop if it is 0 since it is either the first one or not in the ranks
            if (winnerPosition == 0) {
                break;
            }
            if (
                characterRankingPoints[winnerRankingTier[winnerPosition]] >=
                characterRankingPoints[winnerRankingTier[winnerPosition - 1]]
            ) {
```

Reply:
This is not really the case. Your for loop condition is `> 0` meaning it will break the loop if `winnerPosition == 0`. So that condition is qust duplicate code